sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs

```diff
@@ -34,65 +34,40 @@ protected ArmResource()
         /// <param name="id"> The identifier of the resource that is the target of operations. </param>
         protected ArmResource(ArmResource parentOperations, ResourceIdentifier id)
-            : this(new ClientContext(parentOperations.ClientOptions, parentOperations.Credential, parentOperations.BaseUri, parentOperations.Pipeline), id)
+            : this(parentOperations.ArmClient, id)
         {
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ArmResource"/> class.
         /// </summary>
-        /// <param name="clientOptions"></param>
-        /// <param name="credential"></param>
-        /// <param name="uri"></param>
-        /// <param name="pipeline"></param>
+        /// <param name="armClient"> The <see cref="ArmClient"/> this resource client should be created from. </param>
         /// <param name="id"> The identifier of the resource that is the target of operations. </param>
-        protected ArmResource(ArmClientOptions clientOptions, TokenCredential credential, Uri uri, HttpPipeline pipeline, ResourceIdentifier id)
-            : this(new ClientContext(clientOptions, credential, uri, pipeline), id)
-        {
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ArmResource"/> class.
-        /// </summary>
-        /// <param name="clientContext"></param>
-        /// <param name="id"> The identifier of the resource that is the target of operations. </param>
-        internal ArmResource(ClientContext clientContext, ResourceIdentifier id)
+        protected internal ArmResource(ArmClient armClient, ResourceIdentifier id)
         {
             Argument.AssertNotNull(id, nameof(id));
 
-            ClientOptions = clientContext.ClientOptions;
+            ArmClient = armClient;
             Id = id;
-            Credential = clientContext.Credential;
-            BaseUri = clientContext.BaseUri;
-            Pipeline = clientContext.Pipeline;
         }
 
-        private Tenant Tenant => _tenant ??= new Tenant(ClientOptions, Credential, BaseUri, Pipeline);
+        private Tenant Tenant => _tenant ??= new Tenant(ArmClient);
 
         /// <summary>
-        /// Gets the resource identifier.
+        /// Gets the <see cref="ArmClient"/> this resource client was created from.
         /// </summary>
-        public virtual ResourceIdentifier Id { get; }
+        protected internal virtual ArmClient ArmClient { get; }
 
-        /// <summary>
-        /// Gets the Azure Resource Manager client options.
-        /// </summary>
-        protected internal virtual ArmClientOptions ClientOptions { get; private set; }
+        internal DiagnosticsOptions DiagnosticOptions => ArmClient.DiagnosticOptions;
 
-        /// <summary>
-        /// Gets the Azure credential.
-        /// </summary>
-        protected internal virtual TokenCredential Credential { get; private set; }
+        internal HttpPipeline Pipeline => ArmClient.Pipeline;
```

Comment:
ArmClient.Pipeline is already internal. Do we need this proxy property?

Reply:
it just means we have to use ArmClient.Pipeline in LOTS of places.  This is mostly there for a shortcut.

Reply:
can we make the ones on ArmClient internal?

Reply:
breaks the mocking if we do that, unless I am missing something

Reply:
You only need to be able to mock things that are accessible to customers.

Reply:
the mocked client ends up passing null to its downstream clients

Reply:
Do you mean instrumented?